src/actions/braze/braze.ts

```diff
@@ -0,0 +1,189 @@
+import * as req from "request-promise-native"
+import * as url from "url"
+
+import * as Hub from "../../hub"
+
+enum BrazeConfig {
+  EXPORT_PATH = "/users/track",
+  LOOKER_ATTRIBUTE_NAME = "looker_export",
+  MAX_LINES = 75,
+  BRAZE_ID_TAG = "braze_id",
+  EXPORT_DEFAULT_VALUE = "LOOKER_EXPORT",
+  MAX_EXPORT = 10000,
+  DEFAULT_DOMAIN = ".braze.com",
+}
+
+interface BrazeApiRow {
+  [key: string]: any
+  external_id?: string
+  braze_id?: string
+  _update_existing_only: boolean
+  looker_export?: {
+    add: string[],
+  }
+}
+
+interface BrazeApiBody {
+  api_key: string
+  attributes: BrazeApiRow[]
+}
+
+function isEmpty(obj: any) {
+  return !obj || Object.keys(obj).length === 0
+}
+
+export class BrazeAction extends Hub.Action {
+  name = "braze"
+  label = "Braze"
+  description = "Ensure there's a '" + BrazeConfig.BRAZE_ID_TAG + "' field tagged in a results."
+    + " MAX EXPORT: " + BrazeConfig.MAX_EXPORT + "."
+  iconName = "braze/braze.svg"
+  supportedActionTypes = [Hub.ActionType.Query]
+  supportedVisualizationFormattings = [Hub.ActionVisualizationFormatting.Noapply]
+  supportedFormattings = [Hub.ActionFormatting.Unformatted]
+  requiredFields = [{ tag: String(BrazeConfig.BRAZE_ID_TAG) }]
+  usesStreaming = true
+  supportedFormats = [Hub.ActionFormat.JsonDetail]
+
+  params = [{
+    name: "braze_api_key",
+    label: "Braze API Key",
+    required: true,
+    sensitive: true,
+    description: "Braze API Key from " +
+      "https://dashboard.braze.com/app_settings/developer_console/ with users.track permission.",
+  },
+  {
+    name: "braze_api_endpoint",
+    label: "Braze REST API Endpoint",
+    required: true,
+    sensitive: false,
+    description: "Braze REST API endpoint based on the instance location. " +
+      "See: https://www.braze.com/docs/developer_guide/rest_api/basics/#endpoints",
+  }]
+
+  async execute( request: Hub.ActionRequest) {
+    // Check for missing fields
+    if (isEmpty( request.params) ) {
+      throw "Missing config settings."
+    }
+
+    if (!(request.params.braze_api_endpoint )
+      || (request.params.braze_api_endpoint.toLowerCase().indexOf(BrazeConfig.DEFAULT_DOMAIN.toLowerCase() ) < 0 )) {
+      throw "Missing or Bad Endpoint."
+    }
+
+    // Generate endpoint
+    const endpoint = request.params.braze_api_endpoint.trim()
+      .replace("http://", "https://").replace(/\/$/, "") + BrazeConfig.EXPORT_PATH
+
+    if (!endpoint.startsWith("http") ) {
+      throw "Incorrect domain for endpoint."
+    }
+
+    if (!request.params.braze_api_key) {
+      throw "Missing API Key."
+    }
+
+    if (!(request.formParams.braze_key)) {
+      throw "Missing primary Braze key."
+    }
+
+    const exportValue = request.formParams.braze_segment  || String(BrazeConfig.EXPORT_DEFAULT_VALUE)
+    const brazeAttribute = { add : [ exportValue ] }
+    const brazeApiKey = String(request.params.braze_api_key)
+    let totalCnt = 0
```

Comment:
Quick thing: let's rename this `totalCount` please.

Reply:
renamed. Thanks!